tsdb/chunks/head_chunks.go

```diff
@@ -376,7 +376,29 @@ func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr erro
 	if err != nil {
 		return files, errors.Wrap(err, "file stat during last head chunk file repair")
 	}
-	if info.Size() == 0 {
+
+	remove := info.Size() == 0 // Empty file.
+	if info.Size() >= MagicChunksSize {
+		// Check if magic number is 0.
+		f, err := os.Open(files[lastFile])
+		if err != nil {
+			return files, errors.Wrap(err, "open file during last head chunk file repair")
+		}
+
+		buf := make([]byte, MagicChunksSize)
+		_, err = f.Read(buf)
+		if err != nil {
+			return files, errors.Wrap(err, "read file during last head chunk file repair")
```

Comment:
I think it would be nice if the error is more specific: `failed to read magic number during last head chunk file repair`. The target audience of the log message is probably operators, so it's ok to mention magic number.

Reply:
Thanks for the suggestion, makes sense